File: NEsper.Tests/support/util/EventPropertyAssertionUtil.cs
using System;
using System.Collections.Generic;

using net.esper.compat;
using net.esper.events;

using NUnit.Core;
using NUnit.Framework;

namespace net.esper.support.util
{
    public class EventPropertyAssertionUtil
    {
        public static void compare(EventBean[] events, IList<EDictionary<String, Object>> expectedValues)
        {
            if ((expectedValues == null) && (events == null))
            {
                return;
            }
            if (((expectedValues == null) && (events != null)) ||
                 ((expectedValues != null) && (events == null)))
            {
                Assert.IsTrue(false);
            }

            Assert.AreEqual(expectedValues.Count, events.Length);

            for (int i = 0; i < expectedValues.Count; i++)
            {
                compare(events[i], expectedValues[i]);
            }
        }

        public static void compare(IEnumerator<EventBean> iterator, IList<EDictionary<String, Object>> expectedValues)
        {
            List<EventBean> values = new List<EventBean>();
            while (iterator.MoveNext())
            {
                values.Add(iterator.Current);
            }

            try
            {
                iterator.MoveNext();
				Object temp = iterator.Current;
                Assert.Fail();
            }
            catch (InvalidOperationException)
            {
                // Expected exception - next called after MoveNext returned false, for testing
            }

            EventBean[] data = null;
            if (values.Count > 0)
            {
                data = values.ToArray();
            }

            compare(data, expectedValues);
        }


        private static void compare(EventBean _event, EDictionary<String, Object> expected)
        {
            foreach (KeyValuePair<String, Object> entry in expected)
            {
                Object valueExpected = entry.Value;
                Object property = _event[entry.Key];

                Assert.AreEqual(valueExpected, property);
            }
        }
    }
}

File: trunk/esper/src/test/java/net/esper/support/util/EventPropertyAssertionUtil.java
package net.esper.support.util;

import net.esper.event.EventBean;

import java.util.*;

import junit.framework.TestCase;

public class EventPropertyAssertionUtil
{
    public static void compare(EventBean[] events, List<Map<String, Object>> expectedValues)
    {
        if ((expectedValues == null) && (events == null))
        {
            return;
        }
        if ( ((expectedValues == null) && (events != null)) ||
             ((expectedValues != null) && (events == null)) )
        {
            TestCase.assertTrue(false);
        }

        TestCase.assertEquals(expectedValues.size(), events.length);

        for (int i = 0; i < expectedValues.size(); i++)
        {
            compare(events[i], expectedValues.get(i));
        }
    }

    public static void compare(Iterator<EventBean> iterator, List<Map<String, Object>> expectedValues)
    {
        ArrayList<EventBean> values = new ArrayList<EventBean>();
        while (iterator.hasNext())
        {
            values.add(iterator.next());
        }

        try
        {
            iterator.next();
            TestCase.fail();
        }
        catch (NoSuchElementException ex)
        {
            // Expected exception - next called after hasNext returned false, for testing
        }

        EventBean[] data = null;
        if (values.size() > 0)
        {
            data = values.toArray(new EventBean[0]);
        }

        compare(data, expectedValues);
    }


    private static void compare(EventBean event, Map<String, Object> expected)
    {
        for (Map.Entry<String, Object> entry : expected.entrySet())
        {
            Object valueExpected = entry.getValue();
            Object property = event.get(entry.getKey());

            TestCase.assertEquals(valueExpected, property);
        }
    }
}
